#!/usr/bin/env python
PACKAGE = "robot_eye_driver"

import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

nodestate_enum = gen.enum([ gen.const("stop", int_t, 0, "The state is STOP"),
                       gen.const("run", int_t, 1, "The state is RUN"),
                       gen.const("pause", int_t, 2, "The state is PAUSE") ],
                       "An enum to set the state od the publisher node")

gen.add("node_state", int_t, 0, "A state parameter which is edited via an enum", 2, 0, 2, edit_method=nodestate_enum)
gen.add("az_rate", double_t, 0, "Number of rounds per second [max:15(5400 degrees/sec)]", 10., 1., 15.)
gen.add("n_lines", int_t, 0, "Number of horizontal lines (vertical resolution) [min:az_rate]", 100, 1, 1000)
gen.add("averaging", int_t, 0, "'averaging' value range:[1,5]", 1, 1, 5)
gen.add("laser_freq", int_t, 0, "Measurement frequency [range:[1,10000] or 30000]", 10000, 1, 30000)
gen.add("intensity", bool_t, 0, "Enable the streaming of intensity values [can be 'on' only if laser_freq <= 10000]", False)

exit(gen.generate(PACKAGE, "robot_eye_driver", "RobotEyeParameters"))
